build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/snyk/GenerateSnykDependencyGraph.java

```diff
@@ -96,7 +96,22 @@ private Map<String, Object> generateGradleGraphPayload() {
             version.get(),
             firstLevelModuleDependencies
         );
-        return Map.of("meta", FIXED_META_DATA, "depGraphJSON", builder.build(), "target", buildTargetData());
+        return Map.of(
+            "meta",
+            FIXED_META_DATA,
+            "depGraphJSON",
+            builder.build(),
+            "target",
+            buildTargetData(),
+            "targetReference",
+            targetReference.get(),
+            "projectAttributes",
+            projectAttributesData()
+        );
+    }
+
+    private Map<String, List<String>> projectAttributesData() {
+        return Map.of("lifecycle", List.of(version.map(v -> v.endsWith("SNAPSHOT") ? "development" : "production").get()));
```

Comment:
This allows filtering in the snyk UI by project lifecycle

Reply:
Realistically will we _ever_ publish "production" reports? Does it really matter? If so we'd have to integrate this into the release build somehow and I don't think it's really worth doing so.

We may want to check with what other teams are doing here. InfoSec may want release build reports for auditing purposes and if so we'll need to sort out how we do that.